pkg/controllers/execution/execution_controller.go

```diff
@@ -273,8 +273,7 @@ func (c *Controller) updateAppliedCondition(work *workv1alpha1.Work, status meta
 		}
 		updated := &workv1alpha1.Work{}
 		if err = c.Get(context.TODO(), client.ObjectKey{Namespace: work.Namespace, Name: work.Name}, updated); err == nil {
-			// make a copy, so we don't mutate the shared cache
-			work = updated.DeepCopy()
+			work = updated
```

Comment:
Yes, we don't need copy here. but I think we can optimize the code. I think we don't need `updated` variable here.